app/scripts/directives/createSecret.js

```diff
@@ -175,6 +175,23 @@ angular.module("openshiftConsole")
           });
         };
 
+        var updateEditorMode = _.debounce(function(){
+          try {
+            JSON.parse($scope.newSecret.data.dockerConfig);
+            delete $scope.alerts.invalidFormat;
+          } catch (e) {
+            $scope.alerts["invalidFormat"] = {
+              type: "warning",
+              message: "Configuration file should be in JSON format."
+            };
+          }
+        }, 300, {
+          'leading': true,
+          'trailing': false
```

Comment:
With `trailing: false`, I don't think it will catch my last keystrokes, which could have fixed the validation failure.